tests/networkpolicy_test.go

```diff
@@ -0,0 +1,169 @@
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+const (
+	netpolDeploymentName = "nginx-deployment"
+)
+
+func getFakePolicyRequest(policyName, fieldName string) *v1.PostPolicyRequest {
+	return &v1.PostPolicyRequest{
+		Policy: &storage.Policy{
+			Id: "",
+			LifecycleStages: []storage.LifecycleStage{
+				storage.LifecycleStage_DEPLOY,
+			},
+			Name:               policyName,
+			IsDefault:          false,
+			CriteriaLocked:     false,
+			MitreVectorsLocked: false,
+			Severity:           storage.Severity_LOW_SEVERITY,
+			Categories:         []string{"Anomalous Activity"},
+			PolicySections: []*storage.PolicySection{
+				{
+					SectionName: "example",
+					PolicyGroups: []*storage.PolicyGroup{
+						{
+							FieldName:       fieldName,
+							BooleanOperator: 0,
+							Negate:          false,
+							Values: []*storage.PolicyValue{
+								{
+									Value: "true",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		EnableStrictValidation: false,
+	}
+}
+
+func createPolicyIfMissing(t *testing.T, client v1.PolicyServiceClient, req *v1.PostPolicyRequest) (func(), error) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+	p, err := client.PostPolicy(ctx, req)
+	if err != nil {
+		if s, ok := status.FromError(err); ok {
+			if s.Code() == codes.Internal && strings.Contains(s.Message(), "Could not add policy due to name validation") {
+				return func() {}, nil
+			}
+		}
+		return nil, err
+	}
+	return func() {
+		deletePolicy(t, client, p.Id)
+	}, nil
+}
+
+func deletePolicy(t *testing.T, client v1.PolicyServiceClient, id string) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+	_, err := client.DeletePolicy(ctx, &v1.ResourceByID{Id: id})
+	assert.NoError(t, err)
+}
+
+func applyDeployment(t *testing.T) func() {
+	// Create deployment that doesn't have any network policies
+	setupDeploymentFromFile(t, netpolDeploymentName, "yamls/nginx.yaml")
+	return func() {
+		teardownDeploymentFromFile(t, netpolDeploymentName, "yamls/nginx.yaml")
+	}
+}
+
+func getAlertsForPolicy(t *testing.T, client v1.AlertServiceClient) []*storage.ListAlert {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+	var alerts []*storage.ListAlert
+	resp, err := client.ListAlerts(ctx, &v1.ListAlertsRequest{
+		Query: "Policy:Automated Missing Ingress Policy+Namespace:default",
+	})
+
+	if err != nil {
+		t.Errorf("Failed to get alerts: %s", err)
+		t.FailNow()
+	}
+
+	alerts = resp.GetAlerts()
+	return alerts
+}
+
+func applyNetworkPolicy(t *testing.T) func() {
+	// Create deployment that doesn't have any network policies
+	applyFile(t, "yamls/allow-ingress-netpol.yaml")
+	return func() {
+		teardownFile(t, "yamls/allow-ingress-netpol.yaml")
+	}
+}
+
+func Test_GetViolationForIngressPolicy(t *testing.T) {
+
+	if !features.NetworkPolicySystemPolicy.Enabled() {
+		t.Skip("Feature flag disabled")
+	}
```

Comment:
When I run this locally, then it skips all the tests (apparently this check runs agains the local codebase). However, the FF is enabled in the cluster-under-tests. I guess this is not an issue in the CI, but it may confuse a bit for local testing.

(I will be reviewing this PR further now)

Reply:
So I think I found an interesting solution to this problem. It turns out that there is an endpoint that exposes the enabled feature flags! 

Other than checking locally if the feature flag is enabled, the test will reach out to the running central to figure out if the flag is enabled or not. Let me know what you think.